platforms/software/build-init/src/test/groovy/org/gradle/api/tasks/wrapper/WrapperTest.groovy

```diff
@@ -237,9 +237,75 @@ class WrapperTest extends AbstractTaskTest {
         String distributionUrl = wrapper.getDistributionUrl()
 
         then:
-        distributionUrl.contains("\u0131") == false
+        !distributionUrl.contains("\u0131")
 
         cleanup:
         Locale.setDefault(originalLocale)
     }
+
+    private static class VersionResolver {
+
+        private final TextResourceFactory textResourceFactory
+
+        VersionResolver(Project project) {
+            this.textResourceFactory = project.getResources().getText()
+        }
+
+        def resolveDownloadUrl(String version, String urlSuffix) {
+            String versionUrl = DistributionLocator.getBaseUrl() + "/versions/" + urlSuffix
```

Comment:
❓ Does that call real endpoint? Should we mock it by setting base url for this system property: https://github.com/gradle/gradle/blob/a6313ff9592aed71bb34694a9f03e9d37dcdc777/subprojects/core/src/main/java/org/gradle/util/internal/DistributionLocator.java#L32
?